src/lib.rs

```diff
@@ -266,3 +266,71 @@ macro_rules! declare_types {
 
     { } => { };
 }
+
+#[test]
+fn cli_test() {
+    use std::process::Command;
+    let cli_output = if cfg!(target_os = "windows") {
+        Command::new("cmd")
+                .args(&["/C", "Implement this later"])
+                .output()
+                .expect("failed to execute process")
+    } else {
+        Command::new("sh")
+                .arg("-c")
+                .arg("cd cli; npm install; npm run transpile")
+                .output()
+                .expect("failed to execute process")
+    };
+    assert!(cli_output.status.success());
+
+    let cli_test_output = if cfg!(target_os = "windows") {
+        Command::new("cmd")
+                .args(&["/C", "Implement this later"])
+                .output()
+                .expect("failed to execute process")
+    } else {
+        Command::new("sh")
+                .arg("-c")
+                .arg("cd test/cli; npm install; npm run transpile && npm test")
+                .output()
+                .expect("failed to execute process")
+    };
+    assert!(cli_test_output.status.success());
+}
+
+#[test]
+fn static_test() {
+    use std::process::Command;
+    let static_output = if cfg!(target_os = "windows") {
```

Comment:
Let's have this function skip out early if the current Rust version isn't on the nightly channel. Here's a plan for how to do it:

- Add [rustc_version](https://crates.io/crates/rustc_version) as a [dev dependency](http://doc.crates.io/specifying-dependencies.html#development-dependencies) to this crate.
- Add an `extern crate rustc_version;` to the top level, and I think we need to wrap that with `#[cfg(test)]` so it's only included when we're testing.
- Check if [`version_meta().unwrap().channel`](http://kimundi.github.io/rustc-version-rs/rustc_version/index.html) is [`Channel::Nightly`](http://kimundi.github.io/rustc-version-rs/rustc_version/enum.Channel.html), and if not, return early.

(I wish there was a `#[cfg(nightly)]` attribute but [that hasn't happened yet](https://internals.rust-lang.org/t/setting-cfg-nightly-on-nightly-by-default/1893). If it ever does we can simplify this!)